### PR TITLE
syntax highlighting (#88)

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: src/runner/highlighter.ts [hunk 1]
```diff
@@ -17,58 +17,97 @@
...
-        }
+
+      // NUMBERS
+      if (is(token, ts.SyntaxKind.NumericLiteral)) {
+        highlights.push(highlight(token, "constant"));
       }
-    }
+    });
+
+    output.highlights = highlights;
```
Comment (stas-vilchik) [inline src/runner/highlighter.ts]: This is not the best practice to change the function parameter. Instead I'd `return { highlights };` and merge output from all sensors

File: src/runner/highlighter.ts [hunk 2]
```diff
@@ -17,58 +17,97 @@
...
   }
 }
 
-function highlight(node: ts.Node, highlightKind: SonarTypeOfText) {
+function isKeyword(node: ts.Node): boolean {
```
Comment (stas-vilchik) [inline src/runner/highlighter.ts]: it'd be better to put this into `utils/navigation`, closer to the `is` function

File: tests/runner/highlighter.test.ts [hunk 3]
```diff
@@ -18,20 +18,90 @@
...
+     get b() {
+       return this.a;
+     }
+     static foo() {
+       if (cond);
+     }
+     a: string;
+  }`,
+  );
+  expect(result).toContainEqual(token(1, 0, 1, 5, "keyword")); // class
```
Comment (stas-vilchik) [inline tests/runner/highlighter.test.ts]: maybe use `expect(result).toMatchSnapshot()` ?

File: src/runner/highlighter.ts (hunk 2 above)
Comment (vilchik-elena) [inline src/runner/highlighter.ts]: @stas-vilchik I don't think that this function can be reused.

File: src/runner/highlighter.ts (hunk 1 above)
Comment (vilchik-elena) [inline src/runner/highlighter.ts]: @stas-vilchik done

File: tests/runner/highlighter.test.ts (hunk 3 above)
Comment (vilchik-elena) [inline tests/runner/highlighter.test.ts]: @stas-vilchik As discussed, I prefer these assertions to be explicit in test code.

File: src/runner/highlighter.ts [hunk 4]
```diff
@@ -17,58 +17,101 @@
...
 import { SonarSensor } from "./sensor";
 
 export class SyntaxHighlighter implements SonarSensor {
-  public execute(sourceFile: ts.SourceFile, _: any, output: any): void {
-    new HighlighterWalker(output).walk(sourceFile);
-  }
-}
-
-class HighlighterWalker extends tslint.SyntaxWalker {
+  public execute(sourceFile: ts.SourceFile): HighlightedToken[] {
```
Comment (stas-vilchik) [inline src/runner/highlighter.ts]: @vilchik-elena as discussed, simplify this class to a function

File: src/runner/sensor.ts [hunk 5]
```diff
@@ -20,5 +20,6 @@
 import * as ts from "typescript";
 
 export interface SonarSensor {
-  execute(sourceFile: ts.SourceFile, input: any, output: any): void;
+  execute(sourceFile: ts.SourceFile): any;
```
Comment (stas-vilchik) [inline src/runner/sensor.ts]: I don't think this interface is used

**Pull request merged**